blockchain/My-Wallet-V3-iOS
# feat(KYC): IOS-1104 add user authentication API

## Objective

To (loosely) organize the user authentication process for KYC.

## Description

WIP for client authentication - it does not include fully working requests but opening this now to keep later PRs small.

I'm not totally clear on what to pass into the requests yet which is why `KYCCoordinator`'s `authenticate()` is long and incomplete.

This PR also includes adding public key pinning and user agent to `KYCNetworkRequest`.

## How to Test
If you _really_ want to test...
Run `sh scripts/install-js.sh && sh scripts/build-js.sh`.
Add the following to `start()` in `KYCCoordinator`.

    authenticate()
    return

Add the following to `handleWebLogin()` in `AppCoordinator`.

    let kycCoordinator = KYCCoordinator()
    kycCoordinator.start()
    return

Tapping on "Log in to web wallet" in the side menu should show a `Logger` error
## Screenshot

N/A

## Related Information

* Ticket Number: 1104

## Merge Checklist
